pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -160,6 +169,7 @@ func (c *Controller) reconcile(qualifiedName util.QualifiedName) util.Reconcilia
 		if statusRunning {
 			c.stopController(statusKey, statusStopChan)
 		}
+		typeConfig.Status.ObservedGeneration = typeConfig.Generation
```

Comment:
I think the appropriate place to update the observed generation is just before returning ok.  afaik `observedGeneration` is an indication that a controller has _successfully_ processed the generation in question.

Reply:
Yeah, makes sense, I will remove that initialization from that place.